ingestion/src/metadata/ingestion/source/database/dynamodb.py

```diff
@@ -63,59 +68,76 @@ def create(cls, config_dict, metadata_config: OpenMetadataConnection):
     def prepare(self):
         pass
 
-    def next_record(self) -> Iterable[Entity]:
-        try:
-            table_list = list(self.dynamodb.tables.all())
-            if not table_list:
-                return
-            yield from self.ingest_tables()
-        except Exception as err:
-            logger.debug(traceback.format_exc())
-            logger.error(err)
+    def get_database_names(self) -> Iterable[str]:
+        """
+        Default case with a single database.
 
-    def ingest_tables(self, next_tables_token=None) -> Iterable[OMetaDatabaseAndTable]:
-        tables = list(self.dynamodb.tables.all())
-        for table in tables:
-            try:
+        It might come informed - or not - from the source.
+
+        Sources with multiple databases should overwrite this and
+        apply the necessary filters.
+        """
+
+        database_name = "default"
+        yield database_name
+
+    def yield_database(self, database_name: str) -> Iterable[CreateDatabaseRequest]:
+        """
+        From topology.
+        Prepare a database request and pass it to the sink
+        """
+
+        yield CreateDatabaseRequest(
+            name=database_name,
+            service=EntityReference(
+                id=self.context.database_service.id,
+                type="databaseService",
+            ),
+        )
+
+    def get_database_schema_names(self) -> Iterable[str]:
+        """
+        return schema names
+        """
+        if self.service_connection.__dict__.get("databaseSchema"):
```

Comment:
what happens otherwise? We'll need to be able to list down the existing schemas or the execution will stop here